### PR TITLE
Update sync script

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Update sync script

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18022
https://issues.redhat.com/browse/CRW-1589

### CODE REVIEW

File: build/scripts/sync-chectl-to-crwctl.sh [hunk 1]
```diff
@@ -206,6 +206,11 @@ pushd "${TARGETDIR}" >/dev/null
 	mkdir -p "${TARGETDIR}/${d%/*}"
 	sed -r -e "s#DOC_LINK =.+#DOC_LINK = 'https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces/${CRW_VERSION}/'#" -i "${TARGETDIR}/${d}"
 	sed -r -e "s#DOC_LINK_RELEASE_NOTES.+#DOC_LINK_RELEASE_NOTES = 'https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces/${CRW_VERSION}/html/release_notes_and_known_issues/index'#" -i "${TARGETDIR}/${d}"
+
+	# Restore replaced upstream project
+	sed -r -e "s#CHECTL_PROJECT_NAME =.+#CHECTL_PROJECT_NAME = 'chectl'#" -i "${TARGETDIR}/${d}"
```
Comment (nickboldt) [inline build/scripts/sync-chectl-to-crwctl.sh:211]: why does downstream care about upstream chectl?

File: build/scripts/sync-chectl-to-crwctl.sh [hunk 2]
```diff
@@ -85,8 +85,7 @@ pushd "${SOURCEDIR}" >/dev/null
 			-e "/    const (minishiftAddonTasks|msAddonTasks) = new MinishiftAddonTasks\(\)/d" \
 			-e '/.+tasks.add\(helmTasks.+/d' \
 			-e '/.+tasks.add\((minishiftAddonTasks|msAddonTasks).+/d' \
-			-e "s|(const DEFAULT_CHE_IMAGE =).+|\1 'registry.redhat.io/codeready-workspaces/server-rhel8:${CRW_SERVER_TAG}'|g" \
```
Comment (nickboldt) [inline build/scripts/sync-chectl-to-crwctl.sh:88]: default_che_image is no longer a property? DEFAULT_CHE_OPERATOR_IMAGE got renamed and the tag is removed?

File: build/scripts/sync-chectl-to-crwctl.sh (hunk 1 above)
Comment (tolusha) [reply build/scripts/sync-chectl-to-crwctl.sh:211]: Sync script completely replace `chectl`  on `crwctl` but we have to check if current project is `chectl` or downstream one.

File: build/scripts/sync-chectl-to-crwctl.sh (hunk 2 above)
Comment (tolusha) [reply build/scripts/sync-chectl-to-crwctl.sh:88]: We don't need this. That's kind of legacy.
The image must be defined in `operator.yaml` template.
Here is we just need a image name to check for update task.

**Pull request merged**